internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go

```diff
@@ -254,6 +252,12 @@ var _ dynamicScaleDownProcessor = &standardDynamicScaleDownProcessor{}
 // 2. Clusters in deprovisioning and cleanup state are excluded, as clusters into those states don't accept kafka instances anymore.
 // 3. Clusters that are still not ready to accept kafka instance are also excluded from the capacity calculation
 func (p *standardDynamicScaleDownProcessor) ShouldScaleDown() (bool, error) {
+	// check if the cluster is an enterprise cluster
+	if p.isEnterpriseCluster() {
```

Comment:
I am wondering if this is the place where we should be concerned about it.
What about directly not returning it from the calculated streaming units? or filtering through the locator appropriately?

Reply:
all clusters are returned in the streaming unit calculations: enterprise and managed ones.
There is no notion of locator we could ultimately move the logic here https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/aedf21a02b82289bff00df4cd337fd83ec6d2d26/internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go#L89 if people feel like that's the best place. I do not mind either way

Reply:
I think having some concept of locator as in scale up might be nice. I haven't looked in detail how good it would fit in the current scale down structure though. Could you review if it makes sense similarly to scale up?

Reply:
It could probably make sense and the existing logic in  https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/aedf21a02b82289bff00df4cd337fd83ec6d2d26/internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go#L87 which groups similar streaming unit counts of a cluster (a cluster given by cluster id could appear more than once if it support several kafka instance type) and then used here https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/aedf21a02b82289bff00df4cd337fd83ec6d2d26/internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go#L91-L112 could be refactored onto a locator but I prefer to defer that extra refactoring work from this PR.

Reply:
:+1: